### PR TITLE
fix(swagger): Preserve Model/Classes in swagger while applying decorators.

### LINKED ISSUE #70: bug(swagger): When adding processors, swagger processor removes Models/Class definitions.
Swagger processor removes name of models/classes from swagger object. 
Swagger UI generated from the processed schema has `InlineModel` in models instead of actual model name.
Swagger UI.
screen shot 2017-11-30 at 10 21 25

### PR DESCRIPTION
**Pull request by bhaskarmelkani**
fix(swagger): Preserve Model/Classes in swagger while applying decorators.

Closes #70

### CODE REVIEW

File: lib/nodejs/swagger/swagger.js [hunk 1]
```diff
@@ -80,6 +82,50 @@ class Swagger{
...
+   *
+   * Merges dereferenced and referenced swagger objects, the merged object will contain $refs and the referenced object.
+   */
+  merge (){
+    const validatePromise = this._validate(this.swaggerInput);
+    const bundlePromise = this._bundle(this.swaggerInput);
+    const promiseArray = [validatePromise, bundlePromise];
+    const that = this;
+
+    // Merging referenced and deferened swagger object to have ref links and the derefrenced object.
```
Comment (rbarilani) [inline lib/nodejs/swagger/swagger.js]: @bhaskarmelkani typo here

File: lib/nodejs/swagger/swagger.js [hunk 2]
```diff
@@ -80,6 +82,50 @@ class Swagger{
...
+    return Promise.reduce(
+      promiseArray,
+      (aggregate, swagger) => {
+        // return mergeWith(aggregate, swagger);
```
Comment (rbarilani) [inline lib/nodejs/swagger/swagger.js]: @bhaskarmelkani Is this comment relevant?

File: lib/nodejs/swagger/swagger.js [hunk 3]
```diff
@@ -100,6 +146,57 @@ class Swagger{
     return SwaggerParser.bundle(schema);
   }
 
+  /**
+   * Travers swagger object and unmerge keys matched to fn.
```
Comment (rbarilani) [inline lib/nodejs/swagger/swagger.js]: @bhaskarmelkani not really clear to me...

**Comment by bhaskarmelkani** [comment]
👍

**Comment by rbarilani** [comment]
👍

**Pull request merged**